### PR TITLE
fix: betterer commit flow

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves [] 

### Description

I added a custom reporter so we don't break git apps and skip the logos etc, and also avoid a totally silent reporter.
Screenshot 2023-01-26 at 00 10 51

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,10 +10,11 @@
   "main": "index-common.js",
   "scripts": {
     "android": "react-native run-android",
-    "betterer": "betterer --cachePath .cache/betterer",
-    "betterer:all": "yarn betterer",
-    "betterer:precommit": "yarn betterer precommit",
-    "betterer:show": "yarn betterer results",
+    "betterer-orig": "betterer",
+    "betterer-ours": "betterer --reporter betterer-clean-reporter.ts",
```
Comment (pvinis) [inline package.json:14]: this and the one above are different reporters. ours is silent on success, and shows the errors in failure.

if we want to temporarily see the regular full intense reporter, we can use the `-orig` on the scripts below.

File: src/app/Components/LoadFailureView.tsx [hunk 2]
```diff
@@ -6,6 +6,8 @@ import React, { useRef, useState } from "react"
 import { Animated, Easing } from "react-native"
 import { JustifyContentValue } from "./Bidding/Elements/types"
 
+jest.unmock("react-relay")
+
```
Comment (pvinis) [inline src/app/Components/LoadFailureView.tsx:10]:
```suggestion
```

File: .betterer.ts [hunk 3]
```diff
@@ -3,35 +3,52 @@ import { BettererFileTest } from "@betterer/betterer"
...
 
-  "Avoid using test-renderer!": () =>
-    regexp(/renderWithWrappersLEGACY.* from ".*renderWithWrappers"/).include(typescriptTestFiles),
+  "Avoid using test-renderer": () =>
+    regexp(
+      /renderWithWrappersLEGACY.* from ".*renderWithWrappers"/,
+      "We are migrating away from `react-test-renderer`, towards `@testing-library/react-native`."
+    ).include(typescriptTestFiles),
 
-  "Remove all relay unmocks!": () => regexp(/unmock\("react-relay"\)/).include(typescriptTestFiles),
```
Comment (pvinis) [inline .betterer.ts:24]: short-lived, but very helpful :)

File: .betterer.ts [hunk 4]
```diff
@@ -3,35 +3,52 @@ import { BettererFileTest } from "@betterer/betterer"
 
 const typescriptFiles = ["./src/**/*.ts", "./src/**/*.tsx"]
 const typescriptTestFiles = ["./src/**/*.tests.ts", "./src/**/*.tests.tsx"]
-const imageExtensionsToAvoid = ["png", "jpg", "jpeg"]
+// const imageExtensionsToAvoid = ["png", "jpg", "jpeg"]
```
Comment (damassi) [inline .betterer.ts:6]: Can we remove this?

File: .betterer.ts [hunk 5]
```diff
@@ -3,35 +3,52 @@ import { BettererFileTest } from "@betterer/betterer"
...
-  "Finish our strictnes migration!": () =>
-    regexp(/Unsafe legacy code 🚨 Please delete this/).include(typescriptFiles),
+  "Finish our strictness migration": () =>
+    regexp(
+      /Unsafe legacy code 🚨 Please delete this/,
+      "These comments were added when we switched on TypeScript's strict mode, and their number should only ever go down."
+    ).include(typescriptFiles),
 
-  // "Avoid non-webp images!": () =>
+  // "Avoid non-webp images": () =>
```
Comment (damassi) [inline .betterer.ts:27]: This too

File: .circleci/config.yml [hunk 6]
```diff
@@ -253,7 +253,7 @@ jobs:
           name: Update metaphysics
           command: yarn update-metaphysics
 
-  check-pr:
+  check-code-improvements:
```
Comment (damassi) [inline .circleci/config.yml:256]: 👍

File: betterer-clean-reporter.ts [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import { BettererReporter } from "@betterer/betterer"
+import chalk from "chalk"
+
+const createCleanReporter = (): BettererReporter => ({
+  runEnd: (runSummary) => {
+    if (runSummary.delta === null) return
+
+    if (runSummary.delta.diff === 0) {
+      console.log(chalk.gray("Nothing to see here."))
```
Comment (damassi) [inline betterer-clean-reporter.ts:9]: Prob don't need to say anything if its all good, keep it real lean

File: betterer-clean-reporter.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      return
+    }
+    if (runSummary.delta.diff < 0) {
+      console.log(chalk.green("Unwanted things have been removed. Nice!"))
```
Comment (damassi) [inline betterer-clean-reporter.ts:13]: Maybe something like
```suggestion
      console.log(chalk.green("Cleaned up:"))
```

File: betterer-clean-reporter.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+      console.log(`Test: "${chalk.bold(runSummary.name)}"`)
+      return
+    }
+    if (runSummary.delta.diff > 0) {
+      console.log(chalk.red("Unwanted things have been added!"))
```
Comment (damassi) [inline betterer-clean-reporter.ts:18]:
```suggestion
      console.log(chalk.red("Betterer found some issues in the commit:"))
```

Comment (damassi) [reply betterer-clean-reporter.ts:18]: (Trying to figure out how to make "unwanted" feel less stark lol)

File: package.json [hunk 10]
```diff
@@ -10,10 +10,11 @@
   "main": "index-common.js",
   "scripts": {
     "android": "react-native run-android",
-    "betterer": "betterer --cachePath .cache/betterer",
-    "betterer:all": "yarn betterer",
-    "betterer:precommit": "yarn betterer precommit",
-    "betterer:show": "yarn betterer results",
+    "betterer-orig": "betterer",
```
Comment (damassi) [inline package.json:13]:
```suggestion
    "betterer": "betterer",
```

File: package.json (hunk 1 above)
Comment (damassi) [reply package.json:14]:
```suggestion
    "betterer:clean": "betterer --reporter betterer-clean-reporter.ts",
```

File: package.json [hunk 11]
```diff
@@ -10,10 +10,11 @@
...
+    "betterer-ours": "betterer --reporter betterer-clean-reporter.ts",
+    "betterer:all": "yarn betterer-ours",
```
Comment (damassi) [inline package.json:15]:
```suggestion
    "betterer:all": "yarn betterer:clean",
```

Comment (damassi) [reply package.json:15]: But really this one can go away, since its just aliasing another command

File: package.json [hunk 12]
```diff
@@ -10,10 +10,11 @@
...
+    "betterer:show": "yarn betterer-ours results",
```
Comment (damassi) [inline package.json:16]:
```suggestion
    "betterer:show": "yarn betterer:clean results",
```

File: package.json [hunk 13]
```diff
@@ -10,10 +10,11 @@
...
+    "betterer:update!": "yarn betterer-ours --update",
```
Comment (damassi) [inline package.json:17]:
```suggestion
    "betterer:update!": "yarn betterer:clean --update",
```

**Review by damassi — DISMISSED** [review]
Just a couple naming / rephrasing suggestions to make a bit more formal. 👌 on the custom reporter 👍

File: package.json (hunk 10 above)
Comment (damassi) [reply package.json:13]: Or we can just remove since yarn will automatically execute `.bin` scripts from node_modules via `yarn <some-installed-package>`

**Review by gkartalis — COMMENTED** [review]
Thank youuu 🥇

**Review by gkartalis — DISMISSED** [review]

File: .betterer.ts (hunk 4 above)
Comment (pvinis) [reply .betterer.ts:6]: this will be back once we are done with ios 13, so Id like to keep it until then

File: betterer-clean-reporter.ts (hunk 8 above)
Comment (pvinis) [reply betterer-clean-reporter.ts:13]: im not spending more time on this reporter 😅. its either `Cleaned up something` and that's it, or `Nice!`.

File: package.json (hunk 10 above)
Comment (pvinis) [reply package.json:13]: i decided to use a separate one so its very clear when we use our reporter and when we use the original reporter. i feel like that extra bit is helpful? what do you think?

Comment (pvinis) [reply package.json:13]: ok ive simplified a bit

**Comment by pvinis** [comment]
i feel like the scripts now are fine.

- `:all` will do the main thing, manually and in precommit, with the custom clean reporter
- `:update!` will do the same, while forcing the results to update when we have added bad things on purpose
- `:show` is just printing a huge list of all the issues that exist now, using the regular reporter

`betterer-clean` uses `-` and not `:` because its not a subcommand, but a different command.

**Review by gkartalis — DISMISSED** [review]
Let's merge @pvinis ??

**Review by damassi — DISMISSED** [review]
Updates look great 🚀 🙏

File: package.json (hunk 10 above)
Comment (damassi) [reply package.json]: ty! changes look right

**Pull request merged**